api/v1/handlers.go

```diff
@@ -1773,20 +1774,32 @@ type minerResp struct {
 // @Success      200  {object}  string
 // @Failure      400           {object}  util.HttpError
 // @Failure      500           {object}  util.HttpError
-// @Router       /public/miners [get]
+// @Param        params           path      bool  false   "set true to include on-chain miner info"
+// @Router       /public/miners/{lookup_chain} [get]
 func (s *apiV1) handleAdminGetMiners(c echo.Context) error {
+	ctx := context.TODO()
 	var miners []model.StorageMiner
 	if err := s.DB.Find(&miners).Error; err != nil {
 		return err
 	}
 
+	lookupChain, _ := strconv.ParseBool(c.Param("lookup_chain"))
```

Comment:
what do you guys think of this approach?

I think the `GetMinerChainInfo` call is fairly expensive as it looks up the addresses on-chain, so I added a param to prevent it being ran unless necessary

Reply:
we can just use the cache instead of using a param